scipy/integrate/_cubature.py

```diff
@@ -0,0 +1,333 @@
+        `quad_vec`.
+    rtol : float, optional
+        Relative tolerance. Default is 1e-05.
+    atol : float, optional
+        Absolute tolerance. Default is 1e-08.
+    max_subdivisions : int, optional
+        Upper bound on the number of subdivisions to perform to improve the estimate
+        over a subregion. Default is 10,000.
+    args : tuple, optional
+        Additional positional args passed to `f`, if any.
+    kwargs : tuple, optional
+        Additional keyword args passed to `f`, if any.
+    workers : int or map-like callable, optional
+        If `workers` is an integer, part of the computation is done in parallel
+        subdivided to this many tasks (using :class:`python:multiprocessing.pool.Pool`).
+        Supply `-1` to use all cores available to the Process. Alternatively, supply a
+        map-like callable, such as :meth:`python:multiprocessing.pool.Pool.map` for
+        evaluating the population in parallel. This evaluation is carried out as
+        ``workers(func, iterable)``.
+
+    Returns
+    -------
+    res : CubatureResult
+        Result of estimation. See `CubatureResult`.
+
+    Examples
+    --------
+    A simple 1D integral with vector output. Here ``f(x) = x^n`` is integrated over the
+    interval ``[0, 1]``. Since no rule is specified, the default "gk21" is used, which
+    corresponds to `GaussKronrod` rule with 21 nodes.
+
+    >>> import numpy as np
+    >>> from scipy.integrate import cubature
+    >>> def f(x, n):
+    ...    return x.reshape(-1, 1)**n  # Make sure x and n are broadcastable
+    >>> res = cubature(
+    ...     f,
+    ...     a=[0],
+    ...     b=[1],
+    ...     args=(
+    ...         # Since f accepts arrays of shape (npoints, ndim) we need to
+    ...         # make sure n is the right shape
+    ...         np.arange(10).reshape(1, -1),
+    ...     )
+    ... )
+    >>> res.estimate
+     array([1.        , 0.5       , 0.33333333, 0.25      , 0.2       ,
+            0.16666667, 0.14285714, 0.125     , 0.11111111, 0.1       ])
+
+    A 7D integral with arbitrary-shaped array output. Here::
+        f(x) = cos(2*pi*r + alphas @ x)
+
+    for some ``r`` and ``alphas``, and the integral is performed over the unit
+    hybercube, :math:`[0, 1]^7`. Since the integral is in a moderate number of
+    dimensions, "genz-malik" is used rather than the default "gauss-kronrod" to avoid
+    constructing a product rule with :math:`21^7 \approx 2 \times 10^9` nodes.
+
+    >>> import numpy as np
+    >>> from scipy.integrate import cubature
+    >>> def f(x, r, alphas):
+    ...     # f(x) = cos(2*pi*r + alpha @ x)
+    ...     # Need to allow r and alphas to be arbitrary shape
+    ...     npoints, ndim = x.shape[0], x.shape[-1]
+    ...     alphas_reshaped = alphas[np.newaxis, :]
+    ...     x_reshaped = x.reshape(npoints, *([1]*(len(alphas.shape) - 1)), ndim)
+    ...     return np.cos(2*np.pi*r + np.sum(alphas_reshaped * x_reshaped, axis=-1))
+    >>> np.random.seed(1)
```

Comment:
Should use `np.random.default_rng()` and then the doc build will automatically add a seed